dashboard/gen-dashboard.py

```diff
@@ -0,0 +1,133 @@
+            workflow_file = os.path.basename(workflow.path)
+            if workflow_file in exlude_workflows.get(repo.name, []) \
+                or workflow_file == "pr.yml":
+                continue
+            workflow_data[repo.name][workflow_file] = workflow
+
+    return workflow_data
+
+def print_header():
+    print("""---
+title: "Test status"
+SPDX-License-Identifier: CC-BY-SA-4.0
+SPDX-FileCopyrightText: 2022 seL4 Project a Series of LF Projects, LLC.
+---
+
+<!-- Do not edit. This file is generated by seL4/ci-actions/dashboard/gen-dashboard.py -->
+
+# seL4 Test Status
+
+The following list shows the current status of all seL4 GitHub test workflows
+on the main branch.
+
+## Main Tests
+
+<span class="plain-links">[![seL4Test](https://github.com/seL4/seL4/workflows/seL4Test/badge.svg?branch=master)](https://github.com/seL4/seL4/actions/workflows/sel4test-deploy.yml?query=branch%3Amaster)</span>
+<span class="plain-links">[![Proofs](https://github.com/seL4/l4v/workflows/Proofs/badge.svg?branch=master)](https://github.com/seL4/l4v/actions/workflows/proof-deploy.yml?query=branch%3Amaster)</span>
+<span class="plain-links">[![Proof Sync](https://github.com/seL4/seL4/workflows/Proof%20Sync/badge.svg?branch=master)](https://github.com/seL4/seL4/actions/workflows/preprocess-deploy.yml?query=branch%3Amaster)</span>
+<span class="plain-links">[![seL4Bench](https://github.com/seL4/sel4bench/workflows/seL4Bench/badge.svg?branch=master)](https://github.com/seL4/sel4bench/actions/workflows/sel4bench.yml?query=branch%3Amaster)</span>
+<span class="plain-links">[![CAmkES](https://github.com/seL4/camkes-tool/workflows/CAmkES/badge.svg?branch=master)](https://github.com/seL4/camkes-tool/actions/workflows/camkes-deploy.yml?query=branch%3Amaster)</span>
+<span class="plain-links">[![Camkes VM](https://github.com/seL4/camkes-vm-examples/workflows/Camkes%20VM/badge.svg?branch=master)](https://github.com/seL4/camkes-vm-examples/actions/workflows/camkes-vm-deploy.yml?query=branch%3Amaster)</span>
+
+-----
+
+""")
+
+def print_repo(repo_name):
+    print(f"\n### <span class=\"plain-links\">[{repo_name}](https://github.com/seL4/{repo_name}/)</span>\n")
+
+def print_workflow(repo, workflow):
+    base_url = f"https://github.com/seL4/{repo}/actions/workflows"
+    action_url = f"{base_url}/{os.path.basename(workflow.path)}?query=branch%3Amaster"
+    print(f"<span class=\"plain-links\">[![{workflow.name}]({workflow.badge_url}?branch=master)]({action_url})</span>")
+
+def print_badges(workflows: dict):
+    print("## Main repositories\n")
+    for repo in main_repos:
+        if repo not in workflow_data:
+            print(f"WARNING: main repo '{repo}' not found")
+            continue
+        print_repo(repo)
+        for workflow in workflow_data[repo]:
+            print_workflow(repo, workflow_data[repo][workflow])
+
+    print("\n-----\n")
+
+    print("## All repositories\n")
+    ordered_repos = [r for r in sorted(workflow_data.keys())]
```

Comment:
What magic does this do, why can't you just use the list returned by `sorted()` here?

Reply:
Well spotted, that's a left-over from filtering out repos that were already mentioned in main_repos. I decided to duplicate them after all, because I wanted one complete list somewhere. So yes, this can be collapsed just to `sorted(..)`.